src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -0,0 +1,200 @@
+        _BIAlignmentButtons.at(5) = BIAlign_Right();
+        _BIAlignmentButtons.at(6) = BIAlign_BottomLeft();
+        _BIAlignmentButtons.at(7) = BIAlign_Bottom();
+        _BIAlignmentButtons.at(8) = BIAlign_BottomRight();
+    }
+
+    void Profiles::OnNavigatedTo(const NavigationEventArgs& e)
+    {
+        _State = e.Parameter().as<Editor::ProfilePageNavigationState>();
+
+        const auto& colorSchemeMap{ _State.Schemes() };
+        for (const auto& pair : colorSchemeMap)
+        {
+            _ColorSchemeList.Append(pair.Value());
+        }
+
+        const auto& biAlignmentVal{ static_cast<int32_t>(_State.Profile().BackgroundImageAlignment()) };
+        for (const auto& biButton : _BIAlignmentButtons)
+        {
+            biButton.IsChecked(biButton.Tag().as<int32_t>() == biAlignmentVal);
+        }
+    }
+
+    ColorScheme Profiles::CurrentColorScheme()
+    {
+        const auto schemeName{ _State.Profile().ColorSchemeName() };
+        if (const auto scheme{ _State.Schemes().TryLookup(schemeName) })
+        {
+            return scheme;
+        }
+        else
+        {
+            // This Profile points to a color scheme that was renamed or deleted.
+            // Fallback to Campbell.
+            return _State.Schemes().TryLookup(L"Campbell");
+        }
+    }
+
+    void Profiles::CurrentColorScheme(const ColorScheme& val)
+    {
+        _State.Profile().ColorSchemeName(val.Name());
+    }
+
+    fire_and_forget Profiles::BackgroundImage_Click(IInspectable const&, RoutedEventArgs const&)
+    {
+        auto lifetime = get_strong();
+
+        FileOpenPicker picker;
+
+        _State.WindowRoot().TryPropagateHostingWindow(picker); // if we don't do this, there's no HWND for it to attach to
+        picker.ViewMode(PickerViewMode::Thumbnail);
+        picker.SuggestedStartLocation(PickerLocationId::PicturesLibrary);
+        picker.FileTypeFilter().ReplaceAll({ L".jpg", L".jpeg", L".png", L".gif" });
+
+        StorageFile file = co_await picker.PickSingleFileAsync();
+        if (file != nullptr)
+        {
+            BackgroundImage().Text(file.Path());
+        }
+    }
+
+    fire_and_forget Profiles::Commandline_Click(IInspectable const&, RoutedEventArgs const&)
+    {
+        auto lifetime = get_strong();
+
+        FileOpenPicker picker;
+
+        _State.WindowRoot().TryPropagateHostingWindow(picker); // if we don't do this, there's no HWND for it to attach to
+        picker.ViewMode(PickerViewMode::Thumbnail);
+        picker.SuggestedStartLocation(PickerLocationId::ComputerFolder);
+        picker.FileTypeFilter().ReplaceAll({ L".bat", L".exe", L".cmd" });
+
+        StorageFile file = co_await picker.PickSingleFileAsync();
+        if (file != nullptr)
+        {
+            Commandline().Text(file.Path());
+        }
+    }
+
+    fire_and_forget Profiles::StartingDirectory_Click(IInspectable const&, RoutedEventArgs const&)
+    {
+        auto lifetime = get_strong();
```

Comment:
I'll be honest, I straight up took this from the [FilePicker Sample](https://github.com/microsoft/Windows-universal-samples/blob/master/Samples/FilePicker/cppwinrt/Scenario1_SingleFile.cpp). We don't even use `lifetime`. I'll have to play around a bit here to see if that works. I'll get back to you.

Reply:
so, it really doesn't matter. `lifetime` anchors the page so that it stays alive while the button click handler is being run.

That's _a good thing_.

We've gone crazy with the weak references in Terminal. We make them weak, check if they're alive, all the dang time